reference/matrix/dense_kernels.cpp

```diff
@@ -169,6 +170,49 @@ void compute_dot(std::shared_ptr<const ReferenceExecutor> exec,
 GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_DENSE_COMPUTE_DOT_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void convert_to_coo(std::shared_ptr<const ReferenceExecutor> exec,
+                    matrix::Coo<ValueType, IndexType> *result,
+                    const matrix::Dense<ValueType> *source)
+{
+    auto num_rows = result->get_num_rows();
+    auto num_cols = result->get_num_cols();
+    auto num_nonzeros = result->get_num_stored_elements();
+
+    auto row_idxs = result->get_row_idxs();
+    auto col_idxs = result->get_col_idxs();
+    auto values = result->get_values();
+
+    auto idxs = 0;
+    for (size_type row = 0; row < num_rows; ++row) {
+        for (size_type col = 0; col < num_cols; ++col) {
+            auto val = source->at(row, col);
+            if (val != zero<ValueType>()) {
+                row_idxs[idxs] = row;
+                col_idxs[idxs] = col;
+                values[idxs] = val;
+                ++idxs;
+            }
+        }
+    }
+}
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_DENSE_CONVERT_TO_COO_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void move_to_coo(std::shared_ptr<const ReferenceExecutor> exec,
```

Comment:
Don't think there are any important optimizations that can be done when moving from Dense to COO, so we can just remove this kernel, and always call `convert_to_coo` directly.

Reply:
@gflegar 
When I delete those move_to_coo function, compiling is failed.
It seems to need the implementation of move_to function due to ConvertibleTo class.

Reply:
You still need to have `Dense::move_to(Coo *)` method. You just call `convert_to_coo` kernel, instead of `move_to_coo` kernel inside it.

Reply:
Fix it.